geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/GfshParser.java

```diff
@@ -123,18 +123,25 @@ else if (insideQuoteOf == c) {
 
     List<String> furtherSplitWithEquals = new ArrayList<>();
     for (String token : splitWithWhiteSpaces) {
-      // if this token has equal sign, split around the first occurrence of it
       int indexOfFirstEqual = token.indexOf('=');
-      if (indexOfFirstEqual < 0) {
+      int indexOfFirstDoubleQuote = token.indexOf('"');
+      int indexOfFirstSingleQuote = token.indexOf('\'');
+      if (indexOfFirstEqual < 0 || token.startsWith("-D")) {
         furtherSplitWithEquals.add(token);
       } else {
-        String left = token.substring(0, indexOfFirstEqual);
-        String right = token.substring(indexOfFirstEqual + 1);
-        if (left.length() > 0) {
-          furtherSplitWithEquals.add(left);
-        }
-        if (right.length() > 0) {
-          furtherSplitWithEquals.add(right);
+        if (indexOfFirstDoubleQuote == 0 || indexOfFirstSingleQuote == 0) {
+          furtherSplitWithEquals.add(token);
+        } else if ((indexOfFirstDoubleQuote < 0 && indexOfFirstSingleQuote < 0)
+            || (indexOfFirstEqual < indexOfFirstDoubleQuote)
+            || (indexOfFirstEqual < indexOfFirstSingleQuote)) {
+          String left = token.substring(0, indexOfFirstEqual);
+          String right = token.substring(indexOfFirstEqual + 1);
+          if (left.length() > 0) {
+            furtherSplitWithEquals.add(left);
+          }
+          if (right.length() > 0) {
+            furtherSplitWithEquals.add(right);
+          }
         }
       }
     }
```

Comment:
this entire for loop can be simplified to:

```
for (String token : splitWithWhiteSpaces) {
      // do not split with "=" if this part starts with quotes or is part of -D
      if(token.startsWith("'") || token.startsWith("\"") || token.startsWith("-D")) {
        furtherSplitWithEquals.add(token);
        continue;
      }
      // if this token has equal sign, split around the first occurrence of it
      int indexOfFirstEqual = token.indexOf('=');
      if (indexOfFirstEqual < 0) {
        furtherSplitWithEquals.add(token);
        continue;
      }
      String left = token.substring(0, indexOfFirstEqual);
      String right = token.substring(indexOfFirstEqual + 1);
      if (left.length() > 0) {
        furtherSplitWithEquals.add(left);
      }
      if (right.length() > 0) {
        furtherSplitWithEquals.add(right);
      }
    }
```

Reply:
thanks!